integration/tests/cook/test_cli.py

```diff
@@ -896,3 +895,105 @@ def test_submit_with_executor(self):
         cp, uuids = cli.submit('ls', self.cook_url, submit_flags='--executor bogus')
         self.assertEqual(2, cp.returncode, cp.stderr)
         self.assertIn("invalid choice: 'bogus'", cli.decode(cp.stderr))
+
+    def test_kill_fails_with_duplicate_uuids(self):
+        # Duplicate job and group uuid
+        duplicate_uuid = uuid.uuid4()
+        cp, uuids = cli.submit('ls', self.cook_url, submit_flags=f'--uuid {duplicate_uuid} --group {duplicate_uuid}')
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        cp = cli.kill(uuids, self.cook_url)
+        self.assertEqual(1, cp.returncode, cp.stderr)
+        self.assertIn('Unable to kill due to duplicate UUIDs', cli.decode(cp.stderr))
+        self.assertIn('as a job', cli.decode(cp.stderr))
+        self.assertIn('as a job group', cli.decode(cp.stderr))
+
+        # Duplicate job and instance uuid
+        cp = cli.wait(uuids, self.cook_url)
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        cp, jobs = cli.show_json(uuids, self.cook_url)
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        duplicate_uuid = jobs[0]['instances'][0]['task_id']
+        cp, uuids = cli.submit('ls', self.cook_url, submit_flags=f'--uuid {duplicate_uuid}')
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        cp = cli.kill(uuids, self.cook_url)
+        self.assertEqual(1, cp.returncode, cp.stderr)
+        self.assertIn('Unable to kill due to duplicate UUIDs', cli.decode(cp.stderr))
+        self.assertIn('as a job', cli.decode(cp.stderr))
+        self.assertIn('as a job instance', cli.decode(cp.stderr))
+
+        # Duplicate instance and group uuid
+        cp = cli.wait(uuids, self.cook_url)
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        cp, jobs = cli.show_json(uuids, self.cook_url)
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        duplicate_uuid = jobs[0]['instances'][0]['task_id']
+        cp, uuids = cli.submit('ls', self.cook_url, submit_flags=f'--group {duplicate_uuid}')
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        cp = cli.kill([duplicate_uuid], self.cook_url)
+        self.assertEqual(1, cp.returncode, cp.stderr)
+        self.assertIn('Unable to kill due to duplicate UUIDs', cli.decode(cp.stderr))
+        self.assertIn('as a job group', cli.decode(cp.stderr))
+        self.assertIn('as a job instance', cli.decode(cp.stderr))
```

Comment:
What happens when you have a duplicate of the same type (e.g., two jobs) on two different clusters?

Is there a simple way of testing 3 clusters? I'm thinking not, since that would require bringing up a third cluster for all of these tests, which is probably not worth it... If that's the case, I'd suggest adding some unit tests for handling a response with duplicate entries on 3 or more clusters. A unit test for this case would have caught the repeated entries in the `duplicate_uuids` that I flagged above.

Reply:
This is what a duplicate of the same type on two clusters looks like:

```bash
$ uuid=$(uuidgen)
$ cs --cluster dev0 submit --uuid $uuid ls
Attempting to submit on dev0 cluster...
Job submission succeeded on dev0. Your job UUID is:
3c9ea901-664e-47ec-b00f-5cf3ded3b20d
$ cs --cluster dev1 submit --uuid $uuid ls
Attempting to submit on dev1 cluster...
Job submission succeeded on dev1. Your job UUID is:
3c9ea901-664e-47ec-b00f-5cf3ded3b20d
$ cs kill $uuid
Refusing to kill due to duplicate UUIDs.

3c9ea901-664e-47ec-b00f-5cf3ded3b20d is duplicated:
- as a job on dev0
- as a job on dev1

You might need to explicitly set the cluster where you want to kill by using the --cluster flag.
```

Reply:
I also added another test case which duplicates a UUID 3 times - for a job, an instance, and a group. This new test case would have exposed the bug you identified.